plugins/sharding/execute_context_base.rb

```diff
@@ -0,0 +1,87 @@
+module Groonga
+  module Sharding
+    class ExecuteContextBase
```

Comment:
How about `StreamExecuteContext` or something because this isn't suitable for non stream based execution such as `logical_select`?

Reply:
It make sense.
I have renamed it.